runtime/launcher.m

```diff
@@ -558,11 +576,26 @@ int main (int argc, char **argv)
 		for (int i = 1; i < argc; i++)
 			*ptr++ = argv [i];
 		*ptr = NULL;
-
-		rv = mono_main (new_argc, new_argv);
-
+        
+		if (is_extension) {
+	            rv = NSExtensionMain (new_argc, new_argv);
+	        }
+	        else {
+	            rv = mono_main (new_argc, new_argv);
+	        }
+        
 		free (new_argv);
 	}
 
 	return rv;
 }
+
+int main (int argc, char **argv)
+{
+    return xamarin_main (argc, argv, false);
```

Comment:
Actually it looks like all your indentation in this file is spaces, and not tabs (wrong editor configuration maybe?)